app/src/main/java/org/stacks/app/StacksModule.kt

```diff
@@ -16,7 +24,51 @@ class StacksModule {
 
     @Provides
     @Singleton
-    fun flowSharedPreferences(@ApplicationContext appContext: Context) =
-        FlowSharedPreferences(PreferenceManager.getDefaultSharedPreferences(appContext))
+    fun keyGenParameterSpec(): KeyGenParameterSpec {
+        val builder = KeyGenParameterSpec.Builder(
+            DEFAULT_MASTER_KEY_ALIAS,
+            PURPOSE_ENCRYPT or PURPOSE_DECRYPT
+        )
 
+        if (android.os.Build.VERSION.SDK_INT > android.os.Build.VERSION_CODES.P) {
+            builder.setUserPresenceRequired(true)
+        }
+
+        builder.setBlockModes(KeyProperties.BLOCK_MODE_GCM)
+            .setEncryptionPaddings(KeyProperties.ENCRYPTION_PADDING_NONE)
+            .setKeySize(KEY_SIZE)
+
+        return builder.build()
+    }
+
+
+    @Provides
+    @Singleton
+    fun masterKey(
+        @ApplicationContext appContext: Context,
+        keyGenParameterSpec: KeyGenParameterSpec
+    ): MasterKey =
+        MasterKey.Builder(appContext)
+            .setKeyGenParameterSpec(keyGenParameterSpec)
+            .setUserAuthenticationRequired(true, 5)
```

Comment:
What's this '5'?

Reply:
Just make it more clear what it is, putting the value inside a constant, for example.